fe/fe-core/src/main/java/com/starrocks/sql/optimizer/rule/tree/lowcardinality/DecodeCollector.java

```diff
@@ -464,12 +465,17 @@ public DecodeInfo visitPhysicalOlapScan(OptExpression optExpression, DecodeInfo
             }
 
             // Condition 3: the varchar column has collected global dict
-            if (!IDictManager.getInstance().hasGlobalDict(table.getId(), column.getName(), version)) {
+            Column columnObj = table.getColumn(column.getName());
+            if (columnObj == null) {
+                continue;
```

Comment:
Why is there a column here that doesn't exist in the table? Add a comment to address the issue